main/tests/server/src/com/google/refine/tests/model/UrlFetchingTests.java

```diff
@@ -221,4 +227,59 @@ public void testInvalidUrl() throws Exception {
         Assert.assertTrue(ExpressionUtils.isError(project.rows.get(2).getCellValue(newCol)));
     }
 
+    @Test
+    public void testHttpHeaders() throws Exception {
+        Row row0 = new Row(2);
+        row0.setCell(0, new Cell("http://headers.jsontest.com", null));
+        /* 
+        http://headers.jsontest.com is a service which returns the HTTP request headers
+        as JSON. For example:
+        {
+           "X-Cloud-Trace-Context": "579a1a2ee5c778dfc0810a3bf131ba4e/11053223648711966807",
+           "Authorization": "Basic",
+           "Host": "headers.jsontest.com",
+           "User-Agent": "OpenRefine",
+           "Accept": "*"
+        }
+        */
+
+        project.rows.add(row0);
+
+        String userAgentValue =  "OpenRefine";
+        String authorizationValue = "Basic";
```

Comment:
Only Basic Auth for now, right ?

Reply:
hmm, wondering about Proxy-Authorization as well ? https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication perhaps a later pull if folks want it ?

Reply:
Basic auth is prob going to be the most common, but there are other methods that work through the Authorization request header (e.g. http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html)

Would be easy to add the Proxy-Authorization request header as well